kubric/renderer/blender.py

```diff
@@ -293,39 +308,83 @@ def background_transparency(self) -> bool:
   def background_transparency(self, value: bool):
     self.blender_scene.render.film_transparent = value
 
-  def save_state(self, path: PathLike = "scene.blend", pack_textures: bool = True):
-    """Saves the '.blend' blender file to disk.""" 
-    filepath = str(self.scratch_dir / "scene.blend")
-    logger.debug("copying '%s' → '%s'", filepath, path)
+  def save_state(self, path: PathLike=None, pack_textures: bool=True):
+    """Saves the '.blend' blender file to disk.
 
-    with RedirectStream(stream=sys.stdout, filename=self.log_file):
-      # --- save file to the local scratch directory
-      bpy.ops.wm.save_mainfile(filepath=filepath)
-      # --- embed all textures into the blend file
-      if pack_textures: bpy.ops.file.pack_all()
-
-    # --- copy to the target directory (might be bucket)
-    tf.io.gfile.copy(filepath, path, overwrite=True)
-
-  def render(self, verbose=False):
-    """Renders the animation to `self.scratch_dir`; likely postprocessed by self.postprocess."""
-    with RedirectStream(stream=sys.stdout, filename=self.log_file, disabled=verbose):
-        bpy.ops.render.render(animation=True, write_still=False) #TODO: Issue #95
-      
-  def render_still(self, filepath: PathLike = "kubric.png", verbose=False):
-    """Renders a single frame of the scene to filepath."""
-    assert filepath.endswith(".png")
+    If a scratch_dir is provided, the  file is saved there, and the path is optinally specified.   
+    """ 
+    # --- if a scratch_dir was specified, force use it
+    if self.scratch_dir is not None:
```

Comment:
This feels a bit weird. Wouldn't it be nicer to use the scratch-dir as a fallback only? I.e. if a scratch-dir is specified and a path is provided then the path simply takes precedence?